File: app/src/main/java/edu/tjrac/swant/filesystem/CarryPathDialogFragment.java
package edu.tjrac.swant.filesystem;

import android.annotation.SuppressLint;
import android.app.AlertDialog;
import android.app.Dialog;
import android.content.Context;
import android.content.DialogInterface;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.support.design.widget.TextInputEditText;
import android.support.v4.app.DialogFragment;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;

import com.google.gson.Gson;
import com.yckj.baselib.common.base.BaseApplication;
import com.yckj.baselib.util.StringUtils;

import java.io.File;
import java.util.ArrayList;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.Unbinder;
import edu.tjrac.swant.unicorn.R;

import static edu.tjrac.swant.filesystem.Config.SP.CARRAY_JSON;

/**
 * Created by wpc on 2018-08-24.
 */

@SuppressLint("ValidFragment")
public class CarryPathDialogFragment extends DialogFragment {

    @BindView(R.id.et_from) TextInputEditText mEtFrom;
    @BindView(R.id.et_to) TextInputEditText mEtTo;

    Unbinder unbinder;
    String from, to;
    public static CarrySetting setting;

    public static SharedPreferences sp;

    public CarryPathDialogFragment(String from, String to) {
        this.from = from;
        this.to = to;
        initCarrySetting();
    }

    public static void initCarrySetting() {
        if (sp == null) {
            sp = BaseApplication.getInstance().getSharedPreferences(Config.SP.setting, Context.MODE_PRIVATE);
        }
        String carry = sp.getString(CARRAY_JSON, "");
        Log.i(CARRAY_JSON, carry);
        setting = new Gson().fromJson(carry, CarrySetting.class);
    }

    public class CarrySetting {
        ArrayList<CarryInfo> maps;

        public boolean fromContain(String absolutePath) {
            if (maps != null && maps.size() > 0) {
                for (CarryInfo item : maps) {
                    if (item.from.equals(absolutePath)) {
                        return true;
                    }
                }
            }
            return false;
        }

        public boolean toContain(String absolutePath) {
            if (maps != null && maps.size() > 0) {
                for (CarryInfo item : maps) {
                    if (item.to.equals(absolutePath)) {
                        return true;
                    }
                }
            }
            return false;
        }

    }

    class CarryInfo {
        String from;
        String to;
        int options;//0 剪切

        public CarryInfo(String from, String to) {
            this.from = from;
            this.to = to;
        }
    }

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        builder.setTitle("设置整理路径");
        View view = LayoutInflater.from(getActivity()).inflate(R.layout.dialog_carry_path, null);
        unbinder = ButterKnife.bind(this, view);
        mEtFrom.setText(from);
        mEtTo.setText(to);


        builder.setView(view);
        builder.setPositiveButton("确定", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                String from = mEtFrom.getText().toString().trim();
                String to = mEtTo.getText().toString().trim();
                if (StringUtils.isEmpty(from) || StringUtils.isEmpty(to)) {

                } else {
                    if (from.equals(to)) {

                    } else {
                        File f_from = new File(from);
                        File f_to = new File(to);
                        if (f_from.exists() && f_from.isDirectory() && f_to.exists() && f_to.isDirectory()) {


                            if (setting.maps == null) {
                                setting.maps = new ArrayList<>();
                            } else {
//                                if (setting.fromContain(f_from.getAbsolutePath())) {
//                                    T.show
//                                }
                                setting.toContain(from);
                            }
                            setting.maps.add(new CarryInfo(f_from.getAbsolutePath(), f_to.getAbsolutePath()));
                            sp.edit().putString(CARRAY_JSON, new Gson().toJson(setting)).commit();

                        }
                    }
                }
            }
        });
        builder.setNegativeButton("取消", null);
        return builder.create();
    }


    @Override
    public void onDestroyView() {
        super.onDestroyView();
        unbinder.unbind();
    }


}

File: app/src/main/java/edu/tjrac/swant/filesystem/CarryPathDialogFragment.kt
package edu.tjrac.swant.filesystem

import android.annotation.SuppressLint
import android.app.AlertDialog
import android.app.Dialog
import android.content.Context
import android.content.SharedPreferences
import android.os.Bundle
import android.support.v4.app.DialogFragment
import android.util.Log
import android.view.LayoutInflater
import com.google.gson.Gson
import com.yckj.baselib.common.base.BaseApplication
import com.yckj.baselib.util.StringUtils
import edu.tjrac.swant.filesystem.Config.SP.CARRAY_JSON
import edu.tjrac.swant.unicorn.R
import kotlinx.android.synthetic.main.dialog_carry_path.*
import java.io.File
import java.util.*

/**
 * Created by wpc on 2018-08-24.
 */

@SuppressLint("ValidFragment")
class CarryPathDialogFragment(
        //    @BindView(R.id.et_from) TextInputEditText et_from;
        //    @BindView(R.id.et_to) TextInputEditText et_to;

        internal var from: String, internal var to: String) : DialogFragment() {

    init {
        initCarrySetting()
    }

    inner class CarrySetting {
        internal var maps: ArrayList<CarryInfo>? = null

        fun fromContain(absolutePath: String): Boolean {
            if (maps != null && maps!!.size > 0) {
                for (item in maps!!) {
                    if (item.from == absolutePath) {
                        return true
                    }
                }
            }
            return false
        }

        fun toContain(absolutePath: String): Boolean {
            if (maps != null && maps!!.size > 0) {
                for (item in maps!!) {
                    if (item.to == absolutePath) {
                        return true
                    }
                }
            }
            return false
        }

    }

    internal inner class CarryInfo(var from: String, var to: String) {
        var options: Int = 0//0 剪切
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val builder = AlertDialog.Builder(activity)
        builder.setTitle("设置整理路径")
        val view = LayoutInflater.from(activity).inflate(R.layout.dialog_carry_path, null)
        et_from.setText(from)
        et_to.setText(to)


        builder.setView(view)
        builder.setPositiveButton("确定") { dialog, which ->
            val from = et_from.getText().toString().trim()
            val to = et_to.getText().toString().trim()
            if (StringUtils.isEmpty(from) || StringUtils.isEmpty(to)) {

            } else {
                if (from == to) {

                } else {
                    val f_from = File(from)
                    val f_to = File(to)
                    if (f_from.exists() && f_from.isDirectory() && f_to.exists() && f_to.isDirectory()) {


                        if (setting!!.maps == null) {
                            setting!!.maps = ArrayList<CarryInfo>()
                        } else {
                            //                                if (setting.fromContain(f_from.getAbsolutePath())) {
                            //                                    T.show
                            //                                }
                            setting!!.toContain(from)
                        }
                        setting!!.maps!!.add(CarryInfo(f_from.getAbsolutePath(), f_to.getAbsolutePath()))
                        sp!!.edit().putString(CARRAY_JSON, Gson().toJson(setting)).commit()

                    }
                }
            }
        }
        builder.setNegativeButton("取消", null)
        return builder.create()
    }


    override fun onDestroyView() {
        super.onDestroyView()
    }

    companion object {
        var setting: CarrySetting?=null

        var sp: SharedPreferences? = null

        fun initCarrySetting() {
            if (sp == null) {
                sp = BaseApplication.getInstance().getSharedPreferences(Config.SP.setting, Context.MODE_PRIVATE)
            }
            val carry = sp!!.getString(CARRAY_JSON, "")
            Log.i(CARRAY_JSON, carry)
            setting = Gson().fromJson(carry, CarrySetting::class.java)
        }
    }


}
